rad/src/filters/deviation.rs

```diff
@@ -0,0 +1,431 @@
+    let assume_float = |x| {
+        if let RadonTypes::Float(f) = x {
+            f
+        } else {
+            unreachable!()
+        }
+    };
+
+    if !input.is_homogeneous() {
+        return Err(RadError::UnsupportedOpNonHomogeneous {
+            operator: RadonFilters::DeviationStandard.to_string(),
+        });
+    }
+
+    let value = input.value();
+
+    match value.first() {
+        None => Ok(input.clone()),
+        Some(RadonTypes::Float(_)) => {
+            let mean = reducers::average::mean(input)?;
+            let mean_float = assume_float(mean);
+            let std_dev = reducers::deviation::standard(input)?;
+            let std_dev_float = assume_float(std_dev);
+
+            let (keep_min, keep_max) =
+                standard_limits(mean_float.value(), std_dev_float.value(), sigmas_float);
+
+            let mut result = vec![];
+            for item in input.value() {
+                let x = assume_float(item.clone());
+                let xv = x.value();
+                if xv >= keep_min && xv <= keep_max {
+                    result.push(RadonTypes::Boolean(RadonBoolean::from(true)));
+                } else {
+                    result.push(RadonTypes::Boolean(RadonBoolean::from(false)));
+                }
+            }
+
+            Ok(RadonArray::from(result))
+        }
+        Some(RadonTypes::Array(_)) => {
+            let v = transpose(input)?;
+
+            let mut standard_v = vec![];
+            for v2mean in v.value() {
+                if let RadonTypes::Array(v2mean) = v2mean {
+                    standard_v.push(RadonTypes::from(boolean_standard(&v2mean, sigmas_float)?));
+                } else {
+                    unreachable!()
+                }
+            }
+
+            let o = RadonArray::from(standard_v);
+            let ot = transpose(&o)?;
+
+            Ok(ot)
+        }
+        Some(rad_types) => Err(RadError::UnsupportedFilter {
+            inner_type: rad_types.clone().radon_type_name(),
+            filter: RadonFilters::DeviationStandard.to_string(),
+        }),
+    }
+}
+
+fn standard_limits(mean: f64, std_dev: f64, sigmas: f64) -> (f64, f64) {
+    // Keep values between
+    // [mean - sigmas * std_dev, mean + sigmas * std_dev] (inclusive)
+    let keep_min = mean - sigmas * std_dev;
+    let keep_max = mean + sigmas * std_dev;
+
+    (keep_min, keep_max)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::types::float::RadonFloat;
+    use crate::types::string::RadonString;
+    use std::f64;
+
+    #[test]
+    fn test_filter_deviation_standard_no_arg() {
+        let input_f64 = &[1.0, 2.0];
+        let input_vec: Vec<RadonTypes> = input_f64
+            .iter()
+            .map(|f| RadonTypes::Float(RadonFloat::from(*f)))
+            .collect();
+        let input = RadonArray::from(input_vec);
+        let extra_args = vec![];
+
+        let expected = RadError::WrongArguments {
+            input_type: "RadonArray".to_string(),
```

Comment:
apply in the rest of tests

```suggestion
            input_type: RadonArray::radon_type_name(),
```